### PR TITLE
RPE-31: configure code quality checks

### PR DESCRIPTION
**Pull request by lgonczar**

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    </module>
+    <module name="SingleLineJavadoc">
+      <property name="ignoreInlineTags" value="false"/>
+    </module>
+    <module name="EmptyCatchBlock">
+      <property name="exceptionVariableName" value="expected"/>
+    </module>
+    <module name="CommentsIndentation"/>
+  </module>
+</module>
```
Comment (rkondratowicz) [inline checkstyle.xml]: is this the original google style or are there any changes here?

Comment (lgonczar) [reply checkstyle.xml]: I've applied indentation changes to the original. They're now in two different commits.

File: build.gradle [hunk 2]
```diff
@@ -1,16 +1,43 @@
...
-    id 'io.spring.dependency-management' version '1.0.3.RELEASE'
-    id 'org.springframework.boot' version '1.5.6.RELEASE'
+  id 'checkstyle'
+  id 'findbugs'
+  id 'org.sonarqube' version '2.5'
+  id 'jacoco'
+  id 'io.spring.dependency-management' version '1.0.3.RELEASE'
+  id 'org.springframework.boot' version '1.5.6.RELEASE'
+  id 'org.owasp.dependencycheck' version '1.4.5.1'
+  id 'com.github.ben-manes.versions' version '0.15.0'
```
Comment (rkondratowicz) [inline build.gradle]: Perhaps update README.md on how to use those

File: build.gradle [hunk 3]
```diff
@@ -1,16 +1,43 @@
 plugins {
-    id 'io.spring.dependency-management' version '1.0.3.RELEASE'
-    id 'org.springframework.boot' version '1.5.6.RELEASE'
+  id 'checkstyle'
+  id 'findbugs'
```
Comment (timja-kainos) [inline build.gradle]: I don't feel great about findbugs and sonar being recommended here

File: build.gradle [hunk 4]
```diff
@@ -1,16 +1,43 @@
...
 }
 
+group = 'uk.gov.hmcts.reform'
 version = '0.0.1'
+
 sourceCompatibility = 1.8
+targetCompatibility = 1.8
+
+checkstyle.toolVersion = '8.2'
+checkstyle.configFile = new File(rootDir, "checkstyle.xml")
```
Comment (timja-kainos) [inline build.gradle]: unless they fixed the issue you may need to add some more config to fail on warnings.
https://github.com/hmcts/cmc-claim-store/blob/master/build.gradle#L28

File: build.gradle (hunk 3 above)
Comment (lgonczar) [reply build.gradle]: Why?

File: build.gradle (hunk 4 above)
Comment (lgonczar) [reply build.gradle]: done

File: build.gradle (hunk 3 above)
Comment (timja-kainos) [reply build.gradle]: they do the same thing, except findbugs takes ages

File: build.gradle (hunk 2 above)
Comment (lgonczar) [reply build.gradle]: done

File: README.md [hunk 5]
```diff
@@ -1,10 +1,72 @@
...
 
+[![Build Status](https://travis-ci.org/hmcts/spring-boot-template.svg?branch=master)](https://travis-ci.org/hmcts/spring-boot-template)
+
 ## Purpose
 
 The purpose of this template is to speed up the creation of new Spring applications within HMCTS
 and help keep the same standards across multpile teams. If you need to create a new app, you can
 simply use this one as a starting point and build on top of it.
 
+TODO: complete the readme
```
Comment (timja-kainos) [inline README.md]: Is this TODO still needed?

File: build.gradle [hunk 6]
```diff
@@ -1,16 +1,54 @@
...
+
+// https://jeremylong.github.io/DependencyCheck/dependency-check-gradle/configuration.html
+dependencyCheck {
+  // Specifies if the build should be failed if a CVSS score above a specified level is identified.
+  // range of 0-10 fails the build, anything greater and it doesn't fail the build
+  failBuildOnCVSS = System.getProperty('dependencyCheck.failBuild') == 'true' ? 0 : 11
+  suppressionFile = 'dependency-check-suppressions.xml'
+}
+
+tasks.withType(FindBugs) {
```
Comment (timja-kainos) [inline build.gradle]: Not needed

Comment (lgonczar) [reply build.gradle]: Right, removed

File: README.md (hunk 5 above)
Comment (lgonczar) [reply README.md]: Yes, I'm going to add more information to the readme as a part of current story, so that people know how to use this template.

**Pull request merged**